src/script/descriptor.cpp

```diff
@@ -1414,34 +1414,76 @@ enum class ParseScriptContext {
+        std::vector<Span<const char>> to_process;
+        Span<const char> item = split[i];
+        if (item.size() > 0 && item[0] == '<' && item[item.size() - 1] == '>') {
+            if (!allow_multipath) {
+                error = strprintf("Key path value '%s' specifies multipath in a section where multipath is not allowed", std::string(item.begin(), item.end()));
+                return false;
+            }
+            if (out.size() > 1) {
+                error = "Multiple multipath key path specifiers found";
+                return false;
+            }
+            to_process = Split(item.subspan(1, item.size() - 2), ';');
+            if (to_process.size() <= 1) {
+                error = "Multipath key path specifiers must have at least two items";
+                return false;
+            }
+        } else {
+            to_process.push_back(item);
+        }
+        std::vector<uint32_t> multipath_elems;
+        for (auto& elem : to_process) {
+            bool hardened = false;
+            if (elem.size() > 0) {
+                const char last = elem[elem.size() - 1];
+                if (last == '\'' || last == 'h') {
+                    elem = elem.first(elem.size() - 1);
+                    hardened = true;
+                    apostrophe = last == '\'';
+                }
+            }
+            uint32_t p;
+            if (!ParseUInt32(std::string(elem.begin(), elem.end()), &p)) {
+                error = strprintf("Key path value '%s' is not a valid uint32", std::string(elem.begin(), elem.end()));
+                return false;
+            } else if (p > 0x7FFFFFFFUL) {
+                error = strprintf("Key path value %u is out of range", p);
+                return false;
+            }
+            p |= ((uint32_t)hardened) << 31;
+            if (std::find(multipath_elems.begin(), multipath_elems.end(), p) != multipath_elems.end()) {
+                error = strprintf("Duplicated key path value %u in multipath specifier", p);
+                return false;
             }
+            multipath_elems.push_back(p);
         }
-        uint32_t p;
-        if (!ParseUInt32(std::string(elem.begin(), elem.end()), &p)) {
-            error = strprintf("Key path value '%s' is not a valid uint32", std::string(elem.begin(), elem.end()));
-            return false;
-        } else if (p > 0x7FFFFFFFUL) {
-            error = strprintf("Key path value %u is out of range", p);
-            return false;
+        if (multipath_elems.size() == 1) {
+            for (auto& kp : out) {
+                kp.push_back(multipath_elems.at(0));
+            }
+            continue;
+        }
+        while (out.size() != multipath_elems.size()) {
+            out.push_back(out.at(0));
+        }
+        for (size_t j = 0; j < multipath_elems.size(); ++j) {
+            out[j].push_back(multipath_elems.at(j));
         }
-        out.push_back(p | (((uint32_t)hardened) << 31));
     }
+
```

Comment:
I didn't find this implementation so friendly and ended up reimplementing the function in a more readable way (at least for me) https://github.com/furszy/bitcoin-core/commit/84bbef2de7f4d3a697b28f6b9177e20dc974b8cc. All yours if you like it. It contain other small code improvements too.